app/services/books/indexing_strategies/i1/strategy.rb

```diff
@@ -31,6 +31,12 @@ def index(book:, index_name:)
       book.root_book_part.pages.each {|page| index_page(page: page, index_name: index_name) }
     end
 
+    def total_number_of_elements_to_index(book:)
+      num_total = 0
+      book.root_book_part.pages.each{|page| num_total += page.elements(element_classes: DESIRED_ELEMENTS_TO_DOCUMENTS.keys).count }
+      num_total
```

Comment:
This works, can also do `.map{...}.sum`.

Are we worried at all about parsing each page twice? Once here, once during indexing.

Can you also check me to see that pages and root book part is memoized (so we aren't actually downloading the book twice)

Reply:
def pages
      @pages ||= parts.map do |part|
        part.is_a?(Page) ? part : part.pages
      end.flatten
    end